bin/pycbc_hdf5_splitbank

```diff
@@ -0,0 +1,117 @@
+                    dictate how to split the bank. It will be split
+                    equally between all specified files.""")
+parser.add_argument("--output-prefix", default=None,
+                    help="""Prefix to add to the output template bank names,
+                    for example 'sub-bank'. The output file names would
+                    become args.output_prefix#.hdf""")
+parser.add_argument("--force", action="store_true", default=False,
+                    help="Overwrite the given hdf file if it exists. "
+                         "Otherwise, an error is raised.")
+parser.add_argument("--verbose", action="store_true", default=False)
+
+
+args = parser.parse_args()
+
+pycbc.init_logging(args.verbose)
+
+logging.info("loading bank")
+
+tmplt_bank = bank.TemplateBank(args.bank_file)
+
+templates = tmplt_bank.table
+
+if args.output_filenames and args.output_prefix:
+    errMsg="""Either supply specific filenames for all output files via
+    args.output_filenames or supply a common prefix for all output files
+    via args.output_prefix. Cannot supply both"""
+    parser.error(errMsg)
+
+# If an array of filenames
+if args.output_filenames:
+    args.number_of_banks = len(args.output_filenames)
+
+# Split the templates in the bank taken as input into the smaller banks
+
+# If the number of output banks is taken as input calculate the number
+# of templates to be stored per bank
+if args.number_of_banks:
+    num_files = args.number_of_banks
+    num_per_file = int(templates[:].size/num_files)
+
+# If the number of templates per bank is taken as input calculate the
+# number of output banks
+elif args.templates_per_bank:
+    num_per_file = args.templates_per_bank
+    num_files = int(templates[:].size / num_per_file)
+
+for ii in range(num_files):
+    start_idx = ii * num_per_file
+    # The output banks are assigned a fixed length equal to the number
+    # of templates per bank requested by the user or calculated earlier
+    # in the code except for the last bank in which the remaining
+    # templates, if any, are put.
+    if ( ii == (num_files-1)):
+        end_idx = templates[:].size
+    else:
+        end_idx = (ii + 1) * num_per_file
+
+    # Assign a name to the h5py output file to store the ii'th smaller bank
+    if args.output_filenames:
+        outname = output_filenames[ii]
+    elif args.output_prefix:
+        outname = args.output_prefix + str(ii) + '.hdf'
+    else:
+        errMsg = """Cannot resolve the process of assigning names to
+                output files. output-filenames or an output-prefix
+                should be taken as input."""
+
+    # Generate the hdf5 output file for the ii'th sub-bank, which would
+    # be a slice of the input template bank having a start index and
+    # end index as calculated above
+    output = tmplt_bank.write_to_hdf(outname, start_idx, end_idx,
+                                     force=args.force, skip_fields='template_duration')
```

Comment:
Close the output file here (i.e., `output.close()`). Otherwise, you might get an error of too many open file handlers if writing out to a large number of files.